### PR TITLE
C# additions to the CSFLE guide

### PR DESCRIPTION
**Pull request by terakilobyte**
## Pull Request Info

This PR adds the C# driver to the CSFLE guide. It still needs to be updated to include the link to the companion CSFLE project, once that is published.

- [x] Companion Project

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13052

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodborg-staging.corp.mongodb.com/2873211/drivers/docsworker-xlarge/DOCSP-13052/security/client-side-field-level-encryption-guide

### CODE REVIEW

File: source/includes/steps-fle-configure-the-mongodb-client.yaml [hunk 1]
```diff
@@ -70,6 +77,20 @@ content: |
                "key": local_master_key
              }
            }
+     .. tab::
+        :tabid: csharp
+
+        .. code-block:: csharp
+
+           var ksmProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
```
Comment (ccho-mongodb) [inline source/includes/steps-fle-configure-the-mongodb-client.yaml:85]:
```suggestion
           var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
```

File: source/includes/steps-fle-create-data-encryption-key.yaml [hunk 2]
```diff
@@ -213,6 +233,23 @@ content: |
...
 
+     .. tab::
+        :tabid: csharp
+
+        .. code-block:: csharp
+
+           var keyVaultClient = new MongoClient(_connectionString);
+           var clientEncryptionOptions = new ClientEncryptionOptions(
+               keyVaultClient: keyVaultClient,
+               keyVaultNamespace: _keyVaultNamespace,
```
Comment (ccho-mongodb) [inline source/includes/steps-fle-create-data-encryption-key.yaml:244]: Just curious to learn what underscore is used for in _keyVaultNamespace and _connectionString, is it special C#-related syntax that modifies the original variable name somehow?

File: source/includes/steps-fle-create-data-encryption-key.yaml [hunk 3]
```diff
@@ -387,6 +424,49 @@ content: |
...
+     .. tab::
+        :tabid: csharp
+
+        .. code-block:: csharp
+
+
+           var client = new MongoClient("mongodb://localhost:27017");
+           // dataKeyId is the UUID from the previous step that was printed
+           // to the console
+           var dataKeyId = "aae4f3b4-91b6-4cef-8867-3113a6dfb27b"
```
Comment (ccho-mongodb) [inline source/includes/steps-fle-create-data-encryption-key.yaml:436]: Should this be something like `"<paste your dataKeyId value here>"`?

File: source/security/client-side-field-level-encryption-guide.txt [hunk 4]
```diff
@@ -262,6 +262,24 @@ Additional Dependencies
...
+      :tabid: csharp
+
+      .. list-table::
+         :header-rows: 1
+
+         * - Dependency Name
+           - Description
+
+         * - x64 Support
+           - x64 support is required for CSFLE
```
Comment (ccho-mongodb) [inline source/security/client-side-field-level-encryption-guide.txt:275]: Is this a .NET setting? Maybe not necessary, but is there a link that might help explain it to a Jasmin persona or non-C# developer like me?

File: source/security/client-side-field-level-encryption-guide.txt [hunk 5]
```diff
@@ -262,6 +262,24 @@ Additional Dependencies
...
+
+
+         * - `libmongocrypt
+             <https://github.com/mongodb/libmongocrypt#installing-libmongocrypt-on-windows>`__
+           - A Windows build of the `libmongocrypt`
```
Comment (ccho-mongodb) [inline source/security/client-side-field-level-encryption-guide.txt:280]: Would it be useful to specify DLL here?
Also nitpick: according to the style guide, list items should start with caps and end with a period.
I'm certain there was another acceptable version in which you start with lowercase and end without period, but it appears to have been removed from our official style guide.

**Review by ccho-mongodb — APPROVED** [review]
Looks good, a few nitpicks and questions.

File: source/includes/steps-fle-create-data-encryption-key.yaml (hunk 2 above)
Comment (terakilobyte) [reply source/includes/steps-fle-create-data-encryption-key.yaml:244]: I'm not sure. During code review with dima these were suggested. I'm guessing it's just more idiomatic.

File: source/includes/steps-fle-create-data-encryption-key.yaml (hunk 3 above)
Comment (terakilobyte) [reply source/includes/steps-fle-create-data-encryption-key.yaml:436]: Yep, good point.

**Pull request merged**